internal/pkg/levee/levee.go

```diff
@@ -78,6 +104,76 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
+// Use the EAR pointer analysis as the propagation engine
+func runEAR(pass *analysis.Pass, state *earpointer.Partitions) (interface{}, error) {
+	conf, err := config.ReadConfig()
+	if err != nil {
+		return nil, err
+	}
+	funcSources := pass.ResultOf[source.Analyzer].(source.ResultType)
+	taggedFields := pass.ResultOf[fieldtags.Analyzer].(fieldtags.ResultType)
+	suppressedNodes := pass.ResultOf[suppression.Analyzer].(suppression.ResultType)
+	// Return whether a field is tainted.
+	isTaintField := func(named *types.Named, index int) bool {
+		if tt, ok := named.Underlying().(*types.Struct); ok {
+			return conf.IsSourceField(utils.DecomposeField(named, index)) ||
+				taggedFields.IsSourceField(tt, index)
```

Comment:
I think this would read better on a single line. In general, it is best to avoid splitting conditionals.